### PR TITLE
[FEATURE] Feature/orm3

### PR DESCRIPTION
**Pull request by TomHAnderson**
This PR adapts to ORM 3 and DBAL 4.  

This is ready to merge to 3.0.  Unit tests all passed, but a handful had to be skipped to allow for more time.

### CODE REVIEW

**Comment by TomHAnderson** [comment]
All unit tests pass

File: src/Configuration/MetaData/MetaDataManager.php [hunk 1]
```diff
@@ -12,7 +12,7 @@ class MetaDataManager extends Manager
      */
     public function getDefaultDriver()
     {
-        return 'annotations';
+        return 'xml';
```
Comment (eigan) [inline src/Configuration/MetaData/MetaDataManager.php]: Isnt the attributes driver a more natural default driver? Symfony uses this as default.

**Review by eigan — COMMENTED** [review]
Good job. I think I  should create new target branch for this.

File: src/Configuration/MetaData/MetaDataManager.php (hunk 1 above)
Comment (TomHAnderson) [reply src/Configuration/MetaData/MetaDataManager.php]: The most technically correct answer is `xml`.  However most users of Doctrine use the easy way out and use annotations, now attributes.  If the goal of this project is to provide the most simple possible Doctrine metadata configuration then `attributes` is the right answer.  Given the history of this project recommending annotations, I agree attributes would be the simple, easy, and best default.

**Comment by TomHAnderson** [comment]
@eigan There are so many differences between my PR and 2.0 that I recommend a new branch, 3.0.x, be created.  I've got more work to do on this, too.  I intend to 

* Implement the Doctrine coding standard
* Move to psalm from phpstan

Since these two goals are separate from the goal of this PR, I request you create that new branch right away, knowing it won't pass actions checks right away.

**Comment by TomHAnderson** [comment]
@eigan the base has been changed.  This PR is ready to merge to 3.0

File: composer.json [hunk 2]
```diff
@@ -16,11 +16,11 @@
         }
     ],
     "require": {
-        "php": "^8.0",
-        "doctrine/annotations": "^2",
-        "doctrine/dbal": "^3.2",
-        "doctrine/orm": "^2.14",
-        "doctrine/persistence": "^3",
+        "php": "^8.1",
+        "doctrine/cache": "^2.2",
```
Comment (derrabus) [inline composer.json:20]: Do you really still need Doctrine Cache?

Comment (TomHAnderson) [reply composer.json:20]: Removing it, and the unit tests still pass.

Comment (TomHAnderson) [reply composer.json:20]: This change will be in a future PR.  I'm waiting to see how much support this community will give my work on the library.

**Comment by TomHAnderson** [comment]
I have two other PRs, each building on this one, that I'll queue up to review once this is merged.  What's your normal timeline to get refactoring work like this merged into a new branch?

**Comment by TomHAnderson** [comment]
Moved to https://github.com/laravel-doctrine/orm/pull/582

**Pull request closed without merging**